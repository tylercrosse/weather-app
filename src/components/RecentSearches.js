import React from "react";

class RecentSearches extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      showRecent: false
    }
  }
  handleClick = (location) => {
    this.props.fetchForecast(location)
    this.setState({
      showRecent: false
    })
  }
  toggleRecent = () => {
    this.setState({
      showRecent: !this.state.showRecent
    })
  }
  render() {
    const { locations } = this.props;
    const recentSearchItems = Object.values(locations)
      .sort((a, b) => b.time - a.time)
      .slice(0, 3)
      .map(location => {
        return (
          <button
            key={location.id}
            className="recentSearch__item"
            onClick={() => this.handleClick(location)}
          >
            {location.address}
          </button>
        );
      });
    return (
      <div className="recentSearches">
        <div>
          <button className="recentSearches__btn" onClick={this.toggleRecent} >
            {this.state.showRecent ? "Hide Recent" : "Show Recent Locations"}
          </button>
        </div>
        {this.state.showRecent && recentSearchItems}
      </div>
    );
  }
}

export default RecentSearches;
