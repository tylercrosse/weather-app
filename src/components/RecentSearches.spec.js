import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import { RecentSearches } from './RecentSearches';

const setup = () => {
  const props = {
    // locations: {},
    // weather: {},
    // geocode: jest.fn(),
    // fetchForecast: jest.fn(),
  }

  const component = <RecentSearches {...props} />;
  const wrapper = shallow(component);

  return {
    props,
    wrapper
  }
}

describe('<RecentSearches />', () => {
  xit('should render correctly', () => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})
