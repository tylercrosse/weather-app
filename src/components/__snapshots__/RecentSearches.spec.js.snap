// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecentSearches /> should render correctly 1`] = `
<RecentSearches
  fetchForecast={[Function]}
  locations={
    Object {
      "ChIJVTPokywQkFQRmtVEaUZlJRA": Object {
        "address": "Seattle, WA, USA",
        "id": "ChIJVTPokywQkFQRmtVEaUZlJRA",
        "lat": 47.6062095,
        "lng": -122.3320708,
        "time": 1499098513,
      },
      "ChIJW-T2Wt7Gt4kRKl2I1CJFUsI": Object {
        "address": "Washington, DC, USA",
        "id": "ChIJW-T2Wt7Gt4kRKl2I1CJFUsI",
        "lat": 38.9071923,
        "lng": -77.03687070000001,
        "time": 1499061460,
      },
    }
  }
>
  <div
    className="recentSearches"
  >
    <div>
      <button
        className="recentSearches__btn"
        onClick={[Function]}
      >
        Show Recent Locations
      </button>
    </div>
  </div>
</RecentSearches>
`;
